frontend/src/components/dashboard/create/email/ProtectedEmailRecipients.tsx

```diff
@@ -0,0 +1,295 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { useParams } from 'react-router-dom'
+import cx from 'classnames'
+
+import { uploadEncryptedFileToS3 } from 'services/upload.service'
+import {
+  CsvStatus,
+  extractTemplateParams,
+  validateCsv,
+  encryptCsv,
+} from 'services/protectedemail.service'
+import {
+  FileInput,
+  CsvUpload,
+  ErrorBlock,
+  PreviewBlock,
+  PrimaryButton,
+  SampleCsv,
+  TextArea,
+  InfoBlock,
+} from 'components/common'
+import { EmailCampaign } from 'classes'
+import { sendTiming } from 'services/ga.service'
+
+import styles from '../Create.module.scss'
+
+const ProtectedEmailRecipients = ({
+  csvFilename: initialCsvFilename,
+  numRecipients: initialNumRecipients,
+  params,
+  onNext,
+}: {
+  csvFilename: string
+  numRecipients: number
+  params: Array<string>
+  onNext: (changes: Partial<EmailCampaign>, next?: boolean) => void
+}) => {
+  const containerRef = useRef<HTMLAnchorElement>(null)
+  const [content, setContent] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [isUploading, setIsUploading] = useState(false)
+  const [isUploadingToServer, setIsUploadingToServer] = useState(false)
+  const [tempFile, setTempFile] = useState<File>()
+  const [encryptedFile, setEncryptedFile] = useState<File>()
+  const [csvInfo, setCsvInfo] = useState<CsvStatus>({
+    numRecipients: initialNumRecipients,
+    csvFilename: initialCsvFilename,
+  })
+  const [encryptionComplete, setEncryptionComplete] = useState(false)
+  const [isCsvEncrypting, setIsCsvEncrypting] = useState(false)
+  const [sampleParams, setSampleParams] = useState(params)
+  const [triggerDownload, setTriggerDownload] = useState(false)
+  const { id: campaignId } = useParams()
+
+  const { csvFilename, numRecipients = 0 } = csvInfo
+
+  // If campaign properties change, bubble up to root campaign object
+  useEffect(() => {
+    onNext({ csvFilename, numRecipients }, false)
+  }, [csvFilename, numRecipients, onNext])
+
+  function renderCreateMessageB() {
+    return (
+      <>
+        <h2>Create password protected message</h2>
+        <h4>Message B</h4>
+        <p>
+          The content below is what your recipients see after opening their
+          password protected mail using their unique password. You can keep
+          editing until you are ready to encrypt the message.
+        </p>
+        <TextArea
+          highlight={true}
+          placeholder="Enter password protected message here"
+          value={content}
+          onChange={setContent}
+        />
+
+        <div className="separator"></div>
+      </>
+    )
+  }
+
+  async function onDownloadSample(
```

Comment:
a bit too hackish. would prefer passing the template into sampleCsv as a param, together with the protected flag